examples/ssh/opkssh/cli.go

```diff
@@ -0,0 +1,319 @@
+	"github.com/openpubkey/openpubkey/client/providers"
+	"github.com/openpubkey/openpubkey/examples/ssh/sshcert"
+	"github.com/openpubkey/openpubkey/pktoken"
+	"github.com/openpubkey/openpubkey/util"
+	"golang.org/x/crypto/ssh"
+)
+
+var (
+	key              = []byte("NotASecureKey123")
+	clientID         = "184968138938-g1fddl5tglo7mnlbdak8hbsqhhf79f32.apps.googleusercontent.com"
+	requiredAudience = "184968138938-g1fddl5tglo7mnlbdak8hbsqhhf79f32.apps.googleusercontent.com"
+	// The clientSecret was intentionally checked in for the purposes of this example,. It holds no power. Do not report as a security issue
+	clientSecret = "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" // Google requires a ClientSecret even if this a public OIDC App
+	issuer       = "https://accounts.google.com"
+	scopes       = []string{"openid profile email"}
+	redirURIPort = "3000"
+	callbackPath = "/login-callback"
+	redirectURI  = fmt.Sprintf("http://localhost:%v%v", redirURIPort, callbackPath)
+)
+
+// This code is currently intended as an example for how OpenPubkey can secure SSH access.
+func main() {
+	if len(os.Args) < 2 {
+		fmt.Printf("Example SSH key generator using OpenPubkey: command choices are: login, ver")
+		return
+	}
+	command := os.Args[1]
+
+	op := providers.GoogleOp{
+		ClientID:     clientID,
+		ClientSecret: clientSecret,
+		Issuer:       issuer,
+		Scopes:       scopes,
+		RedirURIPort: redirURIPort,
+		CallbackPath: callbackPath,
+		RedirectURI:  redirectURI,
+	}
+
+	switch command {
+	case "login":
+		{
+			if len(os.Args) != 2 {
+				fmt.Println("Invalid number of arguments for login, should be `opkssh login`")
+				os.Exit(1)
+			}
+
+			// If principals is empty the server does not enforce any principal.
+			// The OPK verifier should use policy to make this decision.
+			principals := []string{}
+
+			gqFalse := false
+			alg := jwa.ES256
+
+			signer, err := util.GenKeyPair(alg)
+			if err != nil {
+				fmt.Println(err)
+				os.Exit(1)
+			}
+
+			client := &client.OpkClient{
+				Op: &op,
+			}
+
+			certBytes, seckeySshPem, err := CreateSSHCert(context.Background(), client, signer, alg, gqFalse, principals)
+			if err != nil {
+				fmt.Println(err)
+				os.Exit(1)
+			}
+
+			// Write ssh secret key and public key to filesystem
+			err = WriteKeysToSSHDir(seckeySshPem, certBytes)
+			if err != nil {
+				fmt.Println(err)
+				os.Exit(1)
+			}
+			os.Exit(0)
+		}
+	case "ver":
+		{
+			log(strings.Join(os.Args, " "))
+			policyEnforcer := SimpleFilePolicyEnforcer{
+				PolicyFilePath: "/etc/opk/policy",
+			}
+
+			userArg := os.Args[2]
+			certB64Arg := os.Args[3]
+			typArg := os.Args[4]
+
+			authKey, err := AuthorizedKeysCommand(userArg, typArg, certB64Arg, policyEnforcer.CheckPolicy, &op)
+			if err != nil {
+				log(fmt.Sprint(err))
+				os.Exit(1)
+			} else {
+				fmt.Println(authKey)
+				os.Exit(0)
+			}
+		}
+	default:
+		fmt.Println("Error! Unrecognized command:", command)
+	}
+}
+
+type SimpleFilePolicyEnforcer struct {
```

Comment:
The reason objects go at the top of the file is to set up the reader for understanding what the file is about. I think this object should go into its own file or package.